### DISCUSSION TITLE
opposite visualization and noisy activations issues for CLIP

### DISCUSSION THREAD

**Discussion by JianbangZ** [mlfoundations/open_clip#537] [General]
Hi all,
I saw an interesting paper claiming two issues: opposite visualization and noisy activations for original CLIP model, which affects various downstream task performance and proposed an interesting fix without training
https://github.com/xmed-lab/CLIP_Surgery

Basically the CLIP is recognizing target object by looking at the background, instead of foreground, which indicates wrong relation in self-attention. I played with their demo, I think it's indeed so. And I also tested 2 open_clip models to further test it out. Here are the results.

I use an image and look ['window', 'wall','piano','cat'] 
**CLIP VIT-B/16 Official checkpoint**
![CLIP_window](https://github.com/mlfoundations/open_clip/assets/15835199/2c99e142-87ee-4c8c-89c4-0124284888e2)
![CLIP_wall](https://github.com/mlfoundations/open_clip/assets/15835199/e067ad74-416f-46ea-8ac9-1f16b7b2a5df)
![CLIP_piano](https://github.com/mlfoundations/open_clip/assets/15835199/817244cb-1f60-48fc-a9a0-c218c9658741)
![CLIP_cat](https://github.com/mlfoundations/open_clip/assets/15835199/32c766f3-f479-447f-994b-6612169b9a01)
**OPEN_CLIP VIT-B/16 laion2b_s34b_b88k**
![openclip_window](https://github.com/mlfoundations/open_clip/assets/15835199/effe8637-d2f5-46f8-965a-3bd48552e958)
![openclip_wall](https://github.com/mlfoundations/open_clip/assets/15835199/a05ed7fc-2a21-41bb-b9a8-de0cd4d0e01c)
![openclip_piano](https://github.com/mlfoundations/open_clip/assets/15835199/1800b5e6-ef31-409b-9edd-65a37e69fdfa)
![openclip_cat](https://github.com/mlfoundations/open_clip/assets/15835199/f403d255-6043-45a5-908b-e86629599a85)
**OPEN_CLIP VIT-L/14 commonpool_xl_clip_s13b_b90k**
![openclip_datacamp_cat](https://github.com/mlfoundations/open_clip/assets/15835199/ed2c9108-cd60-4dbc-97f0-cb19a4df919e)

**Comment by mitchellnw** [comment]
Cool!
